dhall/tests/Dhall/Test/QuickCheck.hs

```diff
@@ -432,6 +439,12 @@ normalizingAnExpressionDoesntChangeItsInferredType expression =
     filterOutEmbeds :: Typer a
     filterOutEmbeds _ = Const Sort -- This could be any ill-typed expression.
 
+injectThenInterpretIsIdentity :: (Inject a, Interpret a, Eq a) => a -> Property
```

Comment:
If you add a `Typeable` constraint to this then you can have this generate not only the `Property`, but the description, too:

```haskell
injectThenInterpretIsIdentity a =
    ( "Injecting then Interpreting is identity for " <> show (typeOf a)
    , property
    , QuickCheckTests 100
    )
  where
    property = monadicIO $ do …
```

Reply:
Oooh, tasty :) (no pun intended)